js/modules/k6/experimental/tracing/encoding.go

```diff
@@ -17,3 +17,15 @@ func randHexString(n int) string {
 
 	return string(b)
 }
+
+// chance returns true with a `percentage` chance, otherwise false.
+// the `percentage` argument is expected to be
+// within 0 <= percentage <= 100 range.
+//
+// The chance function works under the assumption that the
+// go rand module has been seeded with a non-deterministic
+// value.
+func chance(percentage int) bool {
+	//nolint:gosec
+	return rand.Intn(100) < percentage
```

Comment:
Using the global `rand` means that we lock across all CPU cores to get access to the same rand.

That actually is a problem in some cases and maybe in thos one - so preferably we should use a per VU :thinking: rand source. There is one in the goja VM, but maybe for this we can have one in the tracing module's module instance?

Reply:
TIL 🤓 I genuinely didn't know. That's a super valuable insight, thanks for that 🙏🏻 
I'll add a rand source to the module's instances indeed 👍🏻